### PR TITLE
Port keywordp to rust

### PR DESCRIPTION
**Pull request by shanavas786**
Bit fields manipulations are hacky

### CODE REVIEW

**Comment by shanavas786** [comment]
Travis fails with
```
remacs::lisp::{{impl}}::from_float':

          /home/travis/build/Wilfred/remacs/rust_src/src/lisp.rs:84: undefined reference to `make_float'

          collect2: error: ld returned 1 exit status
```
#333 fails with same error too

any thoughts ?

**Comment by DavidDeSimone** [comment]
Should be fixed in https://github.com/Wilfred/remacs/pull/340. Looks like a nightly changed was introduced a few days ago that broke something to do with the exportation of no_mangle fns.

File: rust_src/src/multibyte.rs [hunk 1]
```diff
@@ -104,6 +104,11 @@ impl LispStringRef {
     pub fn as_mut_slice(&self) -> &mut [u8] {
         unsafe { slice::from_raw_parts_mut(self.data as *mut u8, self.len_bytes() as usize) }
     }
+
+    #[inline]
+    pub fn char_at(&self, index: ptrdiff_t) -> c_char {
```
Comment (birkenfeld) [inline rust_src/src/multibyte.rs]: I'd propose calling this `byte_at` and returning `u8`.

File: rust_src/src/symbols.rs [hunk 2]
```diff
@@ -81,3 +88,23 @@ fn fmakunbound(symbol: LispObject) -> LispObject {
...
+
+// Define this in Rust to avoid unnecessarily consing up the symbol
+// name.
+
+/// Return t if OBJECT is a keyword.
+/// This means that it is a symbol with a print name beginning with `:'
+/// interned in the initial obarray.
+#[lisp_fn]
+fn keywordp(object: LispObject) -> LispObject {
+    if object.is_symbol() {
```
Comment (birkenfeld) [inline rust_src/src/symbols.rs]: `is_symbol`/`as_symbol_or_error` checks should be replaced by `as_symbol` and pattern matching.

File: rust_src/src/multibyte.rs (hunk 1 above)
Comment (shanavas786) [reply rust_src/src/multibyte.rs]: Thanks, done

File: rust_src/src/symbols.rs (hunk 2 above)
Comment (shanavas786) [reply rust_src/src/symbols.rs]: Thanks, It seems `if let` is more cleaner.

File: rust_src/src/multibyte.rs [hunk 3]
```diff
@@ -104,6 +104,11 @@ impl LispStringRef {
     pub fn as_mut_slice(&self) -> &mut [u8] {
         unsafe { slice::from_raw_parts_mut(self.data as *mut u8, self.len_bytes() as usize) }
     }
+
+    #[inline]
+    pub fn byte_at(&self, index: ptrdiff_t) -> u8 {
+        unsafe { *(self.const_data_ptr()).offset(index) }
```
Comment (birkenfeld) [inline rust_src/src/multibyte.rs]: nit: I don't think it needs the extra parens

**Comment by shanavas786** [comment]
Hope Its good to go :)

**Comment by brotzeit** [comment]
I think all changes were addressed.

**Comment by Wilfred** [comment]
@shanavas786 I've added you as a collaborator, so you have the power to approve other PRs too if you wish :)

**Comment by shanavas786** [comment]
@Wilfred Its my pleasure :). Thanks a lot !!

**Pull request merged**